Auto update failing for tree-sitter-query

Hi, I'm Termux 🤖.

I'm here to help you update your Termux packages.

I've tried to update the [tree-sitter-query](https://github.com/termux/termux-packages/blob/9ee1f2383363215e4ce8c396fcaf136ab60c119f/packages/tree-sitter-query) package, but it failed.

Here's the output of the update script:
<details>
<summary>Show log</summary>
<pre lang="console">
INFO: Updating tree-sitter-query [Current version: 0.6.0]
INFO: package being updated to 0.6.1.
Downloading https://github.com/tree-sitter-grammars/tree-sitter-query/archive/refs/tags/v0.6.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 43412  100 43412    0     0   363k      0 --:--:-- --:--:-- --:--:--  363k

You are about to commit these changes:

--------------------
diff --git a/packages/tree-sitter-query/build.sh b/packages/tree-sitter-query/build.sh
index b5f272ca3f..1cf5cd590a 100644
--- a/packages/tree-sitter-query/build.sh
+++ b/packages/tree-sitter-query/build.sh
@@ -2,9 +2,9 @@ TERMUX_PKG_HOMEPAGE=https://github.com/tree-sitter-grammars/tree-sitter-query
 TERMUX_PKG_DESCRIPTION="TS query grammar for tree-sitter"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
-TERMUX_PKG_VERSION="0.6.0"
+TERMUX_PKG_VERSION="0.6.1"
 TERMUX_PKG_SRCURL=https://github.com/tree-sitter-grammars/tree-sitter-query/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=0fda401a86ed800be15b7209b9942b7fc2e6907a0a9d1a49580306d82ce744f1
+TERMUX_PKG_SHA256=18e158092789e4391668760e9ad1b5b4135c8f483d29163912d29e5619518d40
 TERMUX_PKG_AUTO_UPDATE=true
 TERMUX_PKG_UPDATE_TAG_TYPE='newest-tag'
 TERMUX_PKG_BUILD_IN_SRC=true
--------------------

bump(main/tree-sitter-query): 0.6.1


Not committing to Git!

INFO: Trying to build package.
INFO: Package tree-sitter-query exists in main repo.
Running container 'termux-package-builder' from image 'ghcr.io/termux/package-builder'...
termux - building tree-sitter-query for arch aarch64...
::group::INFO: Fetching repo metadata
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100 13134  100 13134    0     0  84115      0 --:--:-- --:--:-- --:--:-- 84192
[packages-cf.termux.dev-apt-termux-main]: Downloading https://packages-cf.termux.dev/apt/termux-main/dists/stable/Release.gpg
[packages-cf.termux.dev-apt-termux-main]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-main]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100 13120  100 13120    0     0  34562      0 --:--:-- --:--:-- --:--:-- 34526
[packages-cf.termux.dev-apt-termux-x11]: 100 13120  100 13120    0     0  34557      0 --:--:-- --:--:-- --:--:-- 34526
[packages-cf.termux.dev-apt-termux-x11]: Downloading https://packages-cf.termux.dev/apt/termux-x11/dists/x11/Release.gpg
[packages-cf.termux.dev-apt-termux-x11]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-x11]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100 13319  100 13319    0     0  36285      0 --:--:-- --:--:-- --:--:-- 36390
[packages-cf.termux.dev-apt-termux-root]: Downloading https://packages-cf.termux.dev/apt/termux-root/dists/root/Release.gpg
[packages-cf.termux.dev-apt-termux-root]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[packages-cf.termux.dev-apt-termux-root]:                                  Dload  Upload   Total   Spent    Left  Speed
[packages-cf.termux.dev-apt-termux-main]: 
[packages-cf.termux.dev-apt-termux-main]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0   2482      0 --:--:-- --:--:-- --:--:--  2479
[packages-cf.termux.dev-apt-termux-main]: 100   833  100   833    0     0   2482      0 --:--:-- --:--:-- --:--:--  2479
[packages-cf.termux.dev-apt-termux-main]: gpg: Signature made Fri 16 May 2025 12:19:45 PM UTC
[packages-cf.termux.dev-apt-termux-main]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-main]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-main]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: 
[packages-cf.termux.dev-apt-termux-x11]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-x11]: 100   833  100   833    0     0   2366      0 --:--:-- --:--:-- --:--:--  2373
[packages-cf.termux.dev-apt-termux-x11]: gpg: Signature made Fri 16 May 2025 03:44:02 PM UTC
[packages-cf.termux.dev-apt-termux-x11]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-x11]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-x11]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: 
[packages-cf.termux.dev-apt-termux-root]:   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
[packages-cf.termux.dev-apt-termux-root]: 100   833  100   833    0     0   2314      0 --:--:-- --:--:-- --:--:--  2320
[packages-cf.termux.dev-apt-termux-root]: gpg: Signature made Wed 07 May 2025 12:32:36 PM UTC
[packages-cf.termux.dev-apt-termux-root]: gpg:                using RSA key CC72CF8BA7DBFA0182877D045A897D96E57CF20C
[packages-cf.termux.dev-apt-termux-root]: gpg: Good signature from "Termux Releases (Termux automatic builds) <contact@termux.dev>" [ultimate]
[packages-cf.termux.dev-apt-termux-root]: gpg:                 aka "Termux Releases (Termux automatic builds) <contact@termux.org>" [ultimate]
::endgroup::
Downloading tree-sitter-query@0.6.1 source from 'https://github.com/tree-sitter-grammars/tree-sitter-query/archive/refs/tags/v0.6.1.tar.gz' if necessary...
Downloading https://github.com/tree-sitter-grammars/tree-sitter-query/archive/refs/tags/v0.6.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0

100 43412  100 43412    0     0   275k      0 --:--:-- --:--:-- --:--:--  275k
-- Android: Targeting API '24' with architecture 'arm64', ABI 'arm64-v8a', and processor 'aarch64'
-- The C compiler identification is Clang 18.0.3
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/builder/.termux-build/_cache/android-r27c-api-24-v1/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done (0.2s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_CXX_FLAGS
    CMAKE_FIND_ROOT_PATH
    CMAKE_FIND_ROOT_PATH_MODE_LIBRARY
    CMAKE_USE_SYSTEM_LIBRARIES
    DOXYGEN_EXECUTABLE


-- Generating done (0.0s)
-- Build files have been written to: /home/builder/.termux-build/tree-sitter-query/src
[1/3] Generating parser.c
FAILED: src/parser.c /home/builder/.termux-build/tree-sitter-query/src/src/parser.c 
cd /home/builder/.termux-build/tree-sitter-query/src && /usr/bin/tree-sitter generate src/grammar.json --abi=15
thread 'main' panicked at src/generate/render.rs:1647:9:
This version of Tree-sitter can only generate parsers with ABI version 13 - 14, not 15
stack backtrace:
   0:     0x55ddb3ae5f8b - <unknown>
   1:     0x55ddb3b2a270 - <unknown>
   2:     0x55ddb3ada4bd - <unknown>
   3:     0x55ddb3ae5d74 - <unknown>
   4:     0x55ddb3b0de2a - <unknown>
   5:     0x55ddb3b0dacd - <unknown>
   6:     0x55ddb3b0e248 - <unknown>
   7:     0x55ddb3ae636e - <unknown>
   8:     0x55ddb3ae61a6 - <unknown>
   9:     0x55ddb3b0df72 - <unknown>
  10:     0x55ddb37c3cb5 - <unknown>
  11:     0x55ddb3903cb3 - <unknown>
  12:     0x55ddb3875b18 - <unknown>
  13:     0x55ddb3874c45 - <unknown>
  14:     0x55ddb37cad05 - <unknown>
  15:     0x55ddb37c64ea - <unknown>
  16:     0x55ddb37dd563 - <unknown>
  17:     0x55ddb37deee9 - <unknown>
  18:     0x55ddb3b0de64 - <unknown>
  19:     0x55ddb3af3a0b - <unknown>
  20:     0x55ddb37deede - <unknown>
  21:     0x7f74407ee1ca - <unknown>
  22:     0x7f74407ee28b - __libc_start_main
  23:     0x55ddb37c4585 - <unknown>
  24:                0x0 - <unknown>
ninja: build stopped: subcommand failed.
ERROR: failed to build.</pre>
</details>

<hr>
<i>
Above error occured when I last tried to update at 2025-05-16 18:17:08 UTC.<br>
Run ID: <a href="https://github.com/termux/termux-packages/actions/runs/15074548608">15074548608</a><br><br>
<b>Note:</b> Automatic updates will be disabled until this issue is resolved.<br>
</i>

--- a/packages/tree-sitter-query/build.sh
+++ b/packages/tree-sitter-query/build.sh
@@ -2,12 +2,19 @@ TERMUX_PKG_HOMEPAGE=https://github.com/tree-sitter-grammars/tree-sitter-query
 TERMUX_PKG_DESCRIPTION="TS query grammar for tree-sitter"
 TERMUX_PKG_LICENSE="Apache-2.0"
 TERMUX_PKG_MAINTAINER="Joshua Kahn @TomJo2000"
-TERMUX_PKG_VERSION="0.6.0"
+TERMUX_PKG_VERSION="0.6.1"
 TERMUX_PKG_SRCURL=https://github.com/tree-sitter-grammars/tree-sitter-query/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=0fda401a86ed800be15b7209b9942b7fc2e6907a0a9d1a49580306d82ce744f1
+TERMUX_PKG_SHA256=18e158092789e4391668760e9ad1b5b4135c8f483d29163912d29e5619518d40
 TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_BUILD_DEPENDS="tree-sitter-cross-tools"
 TERMUX_PKG_UPDATE_TAG_TYPE='newest-tag'
 TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_pre_configure() {
+	if [[ "${TERMUX_ON_DEVICE_BUILD}" != "true" ]]; then
+		export PATH="$TERMUX_PREFIX"/opt/tree-sitter/cross/bin:$PATH
+	fi
+}
 
 termux_step_post_make_install() {
 	install -d "${TERMUX_PREFIX}"/lib/tree_sitter

